File: ios/Classes/shared/BLETransportConfig.swift
//
// Created by Mateus Polonini Cardoso on 01/02/26.
//
import Foundation
import CoreBluetooth

struct BLETransportConfig {
    let serviceUUID: CBUUID
    let writeCharUUID: CBUUID

    let chunkSize: Int
    let chunkDelayUs: useconds_t

    static let zebraDefault = BLETransportConfig(
        serviceUUID: CBUUID(string: "38EB4A80-C570-11E3-9507-0002A5D5C51B"),
        writeCharUUID: CBUUID(string: "38EB4A82-C570-11E3-9507-0002A5D5C51B"),
        chunkSize: 100,
        chunkDelayUs: 10_000
    )
}

File: android/src/main/kotlin/br/com/bluethermal/blue_thermal_plus/shared/BLETransportConfig.kt
package br.com.bluethermal.blue_thermal_plus.shared
import java.util.UUID

data class BLETransportConfig(
    val serviceUUID: UUID,
    val writeCharUUID: UUID,
    val chunkSize: Int,
    val chunkDelayMs: Long // Android usa milissegundos (Thread.sleep)
) {
    companion object {
        val zebraDefault = BLETransportConfig(
            serviceUUID = UUID.fromString("38EB4A80-C570-11E3-9507-0002A5D5C51B"),
            writeCharUUID = UUID.fromString("38EB4A82-C570-11E3-9507-0002A5D5C51B"),
            chunkSize = 100,
            chunkDelayMs = 10L // 10ms equivale a 10.000us do iOS
        )
    }
}